translates: /docs/hooks-intro.md

Jacco opened the pull request:
New PR to support suggestions

Commit:
commit for new PR

degeens commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+Om dit op te lossen **laten Hooks je een component splitsen in kleinere functies gebaseerd op welke onderdelen gerelateerd zijn (zoals het opzetten van een subscription of het ophalen van data)**, in plaats van splitsing te forceren op basis van lifecycle methoden. Je kunt ook opteren om de lokale state van een component te managen met een reducer, om hem voorspelbaarder te maken.
 
-We'll discuss this more in [Using the Effect Hook](/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns).
+We zullen dit verder bespreken in [De the Effect Hook](/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns).
```
```suggestion
We zullen dit verder bespreken in [De Effect Hook](/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns).
```

degeens commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+### Classes verwarren zowel mensen als machines {#classes-confuse-both-people-and-machines}
 
-In addition to making code reuse and code organization more difficult, we've found that classes can be a large barrier to learning React. You have to understand how `this` works in JavaScript, which is very different from how it works in most languages. You have to remember to bind the event handlers. Without unstable [syntax proposals](https://babeljs.io/docs/en/babel-plugin-transform-class-prop ...
+Behalve dat het hergebruik en code organisatie moeilijke maakt, hebben we gemerkt dat classes ook een groot obstakel kunnen zijn bij het leren van React. Je moet begrijpen hoe `this` werkt in JavaScript, wat heel afwijkend is van hoe dat werkt in de meeste andere talen. Je moet er aan denken om de event handlers te binden. Zonder instabiele [syntax voorstellen](https://babeljs.io/docs/en/babel-pl ...
```
```suggestion
Naast dat het hergebruik en organisatie van code moeilijker maakt, hebben we gemerkt dat classes ook een groot obstakel kunnen zijn bij het leren van React. Je moet begrijpen hoe `this` werkt in JavaScript, wat heel afwijkend is van hoe dat werkt in de meeste andere talen. Je moet er aan denken om de event handlers te binden. Zonder instabiele [syntax voorstellen](https://babeljs.io/docs/en/babel-plugin-transform-class-properties/), is de code erg breedsprakig. Mensen kunnen props, state, en top-down data flow prima begrijpen maar toch worstelen met classes. Het verschil tussen functie- en class componenten in React en wanneer welke te gebruiken leidt tot onenigheid, zelfs onder ervaren React developers.
```
Small suggestions for improvement + typo.

degeens commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+We weten dat React developers gefocused zijn op producten maken en niet de tijd hebben om zich te verdiepen in iedere nieuwe API die wordt uitgebracht. Hooks zijn heel nieuw, en het is misschien beter om te wachten op meer voorbeelden en tutorials, voordat je overweegt ze te leren of te gebruiken.
 
-We also understand that the bar for adding a new primitive to React is extremely high. For curious readers, we have prepared a [detailed RFC](https://github.com/reactjs/rfcs/pull/68) that dives into motivation with more details, and provides extra perspective on the specific design decisions and related prior art.
+We begrijpen ook dat de lat voor het toevoegen van een nieuw principe aan React extreem hoog ligt. Voor nieuwsgierige lezers, hebben we een [uitgebreide RFC](https://github.com/reactjs/rfcs/pull/68) gemaakt die in meer detail duikt in de motivatie, en extra perspectief biedt op de specifieke ontwerpbeslissingen en stand van de techniek.
```
```suggestion
We begrijpen ook dat de lat voor het toevoegen van een nieuw principe aan React extreem hoog ligt. Voor nieuwsgierige lezers hebben we een [uitgebreide RFC](https://github.com/reactjs/rfcs/pull/68) gemaakt die in meer detail duikt in de motivatie, en extra perspectief biedt op de specifieke ontwerpbeslissingen en stand van de techniek.
```

degeens reviewed (changes requested):
Here you go... Let me know if you disagree with something. 😉

In addition to my suggestions, you should indicate that a linked article is English by using (Engels). Feel free to check the glossary & style guidelines for more info.

Thank you for your work so far! 💪

Jacco commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+### Het is moeilijk om logica met state te hergebruiken in componenten{#its-hard-to-reuse-stateful-logic-between-components}
 
-React doesn't offer a way to "attach" reusable behavior to a component (for example, connecting it to a store). If you've worked with React for a while, you may be familiar with patterns like [render props](/docs/render-props.html) and [higher-order components](/docs/higher-order-components.html) that try to solve this. But these patterns require you to restructure your components when you use th ...
+React biedt geen manier om herbruikbaar gedrag aan een component te "knopen" (bijvoorbeeld, om het aan een store te koppelen). Als je al een tijd met React werkt, zou je al bekend kunnen zijn met patronen zoals [render props](/docs/render-props.html) en [higher-order components](/docs/higher-order-components.html) die dat proberen op te lossen. Maar om deze patronen te kunnen gebruiken moet je je ...
```
Deze suggestion heeft geen wijziging.

Jacco commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+React biedt geen manier om herbruikbaar gedrag aan een component te "knopen" (bijvoorbeeld, om het aan een store te koppelen). Als je al een tijd met React werkt, zou je al bekend kunnen zijn met patronen zoals [render props](/docs/render-props.html) en [higher-order components](/docs/higher-order-components.html) die dat proberen op te lossen. Maar om deze patronen te kunnen gebruiken moet je je ...
 
-With Hooks, you can extract stateful logic from a component so it can be tested independently and reused. **Hooks allow you to reuse stateful logic without changing your component hierarchy.** This makes it easy to share Hooks among many components or with the community.
+Met Hooks kun je logica met state extraheren uit een component, zodat die onafhankelijk kan worden getest en hergebruikt. **Hooks laten je logica met state hergebruiken zonder aanpassingen in je componenten hiërarchie.** Dit maakt het eenvoudig om Hooks te delen met componenten of met de community.
```
het slaat op "component", ik heb nagezocht dat het "de component" zou moeten zijn waarnaar je dus zou moeten verwijzen met "die". Denk ik. (Ik worstel er ook meet omdat de meeste literatuur over programmeren het steevast over "het component" heeft.

degeens replied:
If I'm not mistaken, it refers to 'logica met state' and not 'component'. I personally think it feels more natural, but I may be wrong. Maybe we need a third opinion?

Jacco replied:
You are right about it refering to 'logica met state'.

Discussed this one with my wife, she says we are both wrong...

She says it should be "hij" or "deze".

I like "deze" best:
```suggestion
Met Hooks kun je logica met state extraheren uit een component, zodat deze onafhankelijk kan worden getest en hergebruikt. **Hooks laten je logica met state hergebruiken zonder aanpassingen in je componenten hiërarchie.** Dit maakt het eenvoudig om Hooks te delen met componenten of met de community.
```

degeens replied:
I agree with your wife, 'deze' is better indeed. 😁

Jacco commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+Daarnaast, React is er nu zo'n vijf jaar, en we willen ervoor zorgen dat het de volgende vijf jaar relevant blijft. Zoals [Svelte](https://svelte.technology/), [Angular](https://angular.io/), [Glimmer](https://glimmerjs.com/), en anderen laten zien, heeft [ahead-of-time compilation](https://en.wikipedia.org/wiki/Ahead-of-time_compilation) van componenten veel toekomstig potentieel. Vooral als dat ...
 
-To solve these problems, **Hooks let you use more of React's features without classes.** Conceptually, React components have always been closer to functions. Hooks embrace functions, but without sacrificing the practical spirit of React. Hooks provide access to imperative escape hatches and don't require you to learn complex functional or reactive programming techniques.
+Om deze problemen op te lossen, **laten Hooks je meer van React's voorzieningen gebruiken zonder classes.** Conceptueel gezien zijn React componenten altijd al meer een soort functies geweest. Hooks omarmen functies, maar zonder de praktische aard van React op te offeren. Hooks bieden toegang to imperatieve ontsnappingsluiken en vereisen niet dat je complexe functional of reactive programmeertech ...
```
Het zou dan denk ik "aangeleund tegen" zijn.

Jacco replied:
Ik stel dit dus voor

```suggestion
Om deze problemen op te lossen, **laten Hooks je meer van React's features gebruiken zonder classes.** Conceptueel gezien hebben React componenten altijd al dichter aangeleund tegen  functies. Hooks omarmen functies, maar zonder de praktische aard van React op te offeren. Hooks bieden toegang tot imperatieve ontsnappingsluiken en vereisen niet dat je complexe functional of reactive programmeertechnieken leert.
```

degeens replied:
Good point. I agree with your suggestion above.

Commit:
Apply suggestions from code review

Thanks for reading through @degeens :-)

I have accepted most. Only few need more discussion :-)

Co-Authored-By: Stijn Geens <33414262+degeens@users.noreply.github.com>

Commit:
Apply suggestions from code review

Zag eerst de wijziging niet. "hechten" vindt ik ook niet heel mooi :S

Co-Authored-By: Stijn Geens <33414262+degeens@users.noreply.github.com>

Jacco commented on content/docs/nav.yml:
```diff
@@ -104,21 +104,21 @@
+        title: Regels van Hooks
       - id: hooks-custom
-        title: Building Your Own Hooks
+        title: Bouw Je Eigen Hooks
```
Ik ga de vertalingen verwijderen. Goed idee

Commit:
Undo menu translations

Commit:
Merge branch 'hook-intro-menu' of https://github.com/Jacco/nl.reactjs.org into hook-intro-menu

Commit:
Apply suggestions from code review

degeens commented on content/docs/nav.yml:
```diff
@@ -108,7 +108,7 @@
   items:
       - id: hooks-intro
-        title: Introducing Hooks
+        title: Introducie tot Hooks
```
```suggestion
        title: Introductie tot Hooks
```
Typo

Commit:
Update content/docs/nav.yml

Co-Authored-By: Stijn Geens <33414262+degeens@users.noreply.github.com>

Commit:
Update content/docs/hooks-intro.md

Commit:
Engels for external links

Commit:
Merge branch 'hook-intro-menu' of https://github.com/Jacco/nl.reactjs.org into hook-intro-menu

degeens commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+We hebben vaak componenten moeten onderhouden die eenvoudig begonnen, maar uitgroeiden tot een ononderhoudbare brei van logica met state en neveneffecten. Iedere lifecycle methode bevat vaak een mix van ongerelateerde logica. Bijvoorbeeld, componenten kunnen het ophalen van data uitvoeren in `componentDidMount` en `componentDidUpdate`. Echter, dezelfde `componentDidMount` methode kan ook ongerela ...
 
-In many cases it's not possible to break these components into smaller ones because the stateful logic is all over the place. It's also difficult to test them. This is one of the reasons many people prefer to combine React with a separate state management library. However, that often introduces too much abstraction, requires you to jump between different files, and makes reusing components more d ...
+In veel gevallen is het niet mogelijk om deze componenten op te splitsen in kleinere componenten omdat de logica met state overal verspreid is. Het is ook moeilijk om ze te testen. Dit is één van de redenen waarom veel mensen verkiezen om React te combineren met een aparte state management library. Maar dat introduceert vaak te veel abstractie, je moet steeds tussen vershillende bestanden heen en ...
```
```suggestion
In veel gevallen is het niet mogelijk om deze componenten op te splitsen in kleinere componenten omdat de logica met state overal verspreid is. Het is ook moeilijk om ze te testen. Dit is één van de redenen waarom veel mensen verkiezen om React te combineren met een aparte state management library. Maar dat introduceert vaak te veel abstractie, je moet steeds tussen verschillende bestanden heen en weer springen, en het maakt hergebruik van componenten moeilijker.
```

degeens commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+### Classes verwarren zowel mensen als machines {#classes-confuse-both-people-and-machines}
 
-In addition to making code reuse and code organization more difficult, we've found that classes can be a large barrier to learning React. You have to understand how `this` works in JavaScript, which is very different from how it works in most languages. You have to remember to bind the event handlers. Without unstable [syntax proposals](https://babeljs.io/docs/en/babel-plugin-transform-class-prop ...
+Naast het moeilijker maken van hergebruik en organisatie van code, hebben we gemerkt dat classes ook een groot obstakel kunnen zijn bij het leren van React. Je moet begrijpen hoe `this` werkt in JavaScript, wat heel afwijkend is van hoe dat werkt in de meeste andere talen. Je moet er aan denken om de event handlers te binden. Zonder instabiele [syntax voorstellen (Engels)](https://babeljs.io/docs ...
```
```suggestion
Naast het moeilijker maken van hergebruik en organisatie van code, hebben we gemerkt dat classes ook een groot obstakel kunnen zijn bij het leren van React. Je moet begrijpen hoe `this` werkt in JavaScript, wat heel afwijkend is van hoe dat werkt in de meeste andere talen. Je moet er aan denken om de event handlers te binden. Zonder instabiele [syntax voorstellen (Engels)](https://babeljs.io/docs/en/babel-plugin-transform-class-properties/), is de code erg breedsprakig. Mensen kunnen props, state en top-down data flow prima begrijpen maar toch worstelen met classes. Het verschil tussen functie- en class componenten in React en wanneer welke te gebruiken leidt tot onenigheid, zelfs onder ervaren React developers.
```
Unnecessary comma.

Commit:
Apply suggestions from code review

Co-Authored-By: Stijn Geens <33414262+degeens@users.noreply.github.com>

Jacco commented:
@Keraito can you also review?

Keraito commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+We hebben vaak componenten moeten onderhouden die eenvoudig begonnen, maar uitgroeiden tot een ononderhoudbare brei van logica met state en neveneffecten. Iedere lifecycle methode bevat vaak een mix van ongerelateerde logica. Bijvoorbeeld, componenten kunnen het ophalen van data uitvoeren in `componentDidMount` en `componentDidUpdate`. Echter, dezelfde `componentDidMount` methode kan ook ongerela ...
 
-In many cases it's not possible to break these components into smaller ones because the stateful logic is all over the place. It's also difficult to test them. This is one of the reasons many people prefer to combine React with a separate state management library. However, that often introduces too much abstraction, requires you to jump between different files, and makes reusing components more d ...
+In veel gevallen is het niet mogelijk om deze componenten op te splitsen in kleinere componenten omdat de logica met state overal verspreid is. Het is ook moeilijk om ze te testen. Dit is één van de redenen waarom veel mensen verkiezen om React te combineren met een aparte state management library. Maar dat introduceert vaak te veel abstractie, je moet steeds tussen verschillende bestanden heen e ...
```
```suggestion
In veel gevallen is het niet mogelijk om deze componenten op te splitsen in kleinere componenten omdat de logica met state overal verspreid is. Het is ook moeilijk om ze te testen. Dit is één van de redenen waarom veel mensen ervoor kiezen om React te combineren met een aparte state management library, maar dat introduceert vaak te veel abstractie. Je moet constant tussen verschillende bestanden heen en weer springen en het maakt hergebruik van componenten moeilijker.
```

Keraito commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+In veel gevallen is het niet mogelijk om deze componenten op te splitsen in kleinere componenten omdat de logica met state overal verspreid is. Het is ook moeilijk om ze te testen. Dit is één van de redenen waarom veel mensen verkiezen om React te combineren met een aparte state management library. Maar dat introduceert vaak te veel abstractie, je moet steeds tussen verschillende bestanden heen e ...
 
-To solve this, **Hooks let you split one component into smaller functions based on what pieces are related (such as setting up a subscription or fetching data)**, rather than forcing a split based on lifecycle methods. You may also opt into managing the component's local state with a reducer to make it more predictable.
+Om dit op te lossen **laten Hooks je een component opsplitsen in kleinere functies gebaseerd op welke onderdelen gerelateerd zijn (zoals het opzetten van een subscription of het ophalen van data)**, in plaats van splitsing te forceren op basis van lifecycle methoden. Je kunt ook opteren om de lokale state van een component te managen met een reducer, om hem voorspelbaarder te maken.
```
```suggestion
Om dit op te lossen **laten Hooks je een component opsplitsen in kleinere functies gebaseerd op welke onderdelen gerelateerd zijn (zoals het opzetten van een subscription of het ophalen van data)**, in plaats van splitsingen te forceren op basis van lifecycle methoden. Je kunt ook ervoor kiezen om de lokale state van een component te managen met een reducer om hem voorspelbaarder te maken.
```

Jacco replied:
```suggestion
Om dit op te lossen **laten Hooks je een component opsplitsen in kleinere functies gebaseerd op welke onderdelen gerelateerd zijn (zoals het opzetten van een subscription of het ophalen van data)**, in plaats van splitsingen te forceren op basis van lifecycle methoden. Je kunt er ook voor kiezen om de lokale state van een component te managen met een reducer, om hem voorspelbaarder te maken.
```

Keraito commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+### Classes verwarren zowel mensen als machines {#classes-confuse-both-people-and-machines}
 
-In addition to making code reuse and code organization more difficult, we've found that classes can be a large barrier to learning React. You have to understand how `this` works in JavaScript, which is very different from how it works in most languages. You have to remember to bind the event handlers. Without unstable [syntax proposals](https://babeljs.io/docs/en/babel-plugin-transform-class-prop ...
+Naast het moeilijker maken van hergebruik en organisatie van code, hebben we gemerkt dat classes ook een groot obstakel kunnen zijn bij het leren van React. Je moet begrijpen hoe `this` werkt in JavaScript, wat heel afwijkend is van hoe dat werkt in de meeste andere talen. Je moet er aan denken om de event handlers te binden. Zonder instabiele [syntax voorstellen (Engels)](https://babeljs.io/docs ...
```
```suggestion
Naast het moeilijker maken van hergebruik en organisatie van code hebben we gemerkt dat classes ook een groot obstakel kunnen zijn bij het leren van React. Je moet begrijpen hoe `this` werkt in JavaScript, wat heel afwijkend is van hoe dat werkt in de meeste andere talen. Je moet er aan denken om de event handlers te binden. Zonder instabiele [syntax voorstellen (Engels)](https://babeljs.io/docs/en/babel-plugin-transform-class-properties/) is de code erg breedsprakig. Mensen kunnen props, state en top-down data flow prima begrijpen maar toch worstelen met classes. Het verschil tussen functie- en class componenten in React en wanneer je welke moet gebruiken leidt tot onenigheid, zelfs onder ervaren React developers.
```

Keraito reviewed (approved):
Looks good in general, awesome work! I've got a few comments and also suggestions, so please look into those!

Commit:
Apply suggestions from code review

Co-Authored-By: Chak Shun Yu <chakshunyu@gmail.com>

Jacco commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+## Geen Breaking Changes {#no-breaking-changes}
 
-Before we continue, note that Hooks are:
+Voor we verder gaan, Hooks zijn:
```
```suggestion
Merk voordat we verder gaan op: Hooks zijn:
```

Commit:
Apply suggestions from code review

Jacco commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+>React 16.8.0 is de eerste release die Hooks ondersteunt. Vergeet niet alle packages te updaten als je upgrade, ook React DOM. React Native zal Hooks ondersteunen in de volgende stable release.
 
-## Video Introduction {#video-introduction}
+## Introductievideo {#video-introduction}
```
I think so.

https://www.linguee.nl/engels-nederlands/vertaling/introductory+video.html

Jacco commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+      <p>Je hebt {count} keer geklikt</p>
       <button onClick={() => setCount(count + 1)}>
-        Click me
+        Klik op me
```
Van het spaans:
https://es.reactjs.org/docs/hooks-intro.html

En onze oosterburen:
https://de.reactjs.org/docs/hooks-intro.html

Spaans dus wel comments niet display waarden.
Duits wel beide vertaald.

Ik vind vertalen beter, tenzij haar het voorbeeld wordt verwezen in een "Pen"

Jacco commented on content/docs/hooks-intro.md:
```diff
@@ -1,110 +1,110 @@
+Hooks lossen een breed scala van schijnbaar ongerelateerde problemen op in React die we tegen zijn gekomen gedurende de vijf jaar dat we tienduizenden componenten geschreven en onderhouden hebben. Of je nu React aan het leren bent, dagelijks gebruikt, of zelfs een andere library prefereert met een vergelijkbaar component model, je zou sommige van deze problemen kunnen herkennen.
 
-### It's hard to reuse stateful logic between components {#its-hard-to-reuse-stateful-logic-between-components}
+### Het is moeilijk om logica met state te hergebruiken in componenten {#its-hard-to-reuse-stateful-logic-between-components}
```
https://github.com/reactjs/nl.reactjs.org/issues/5

But this heading didn't have capitals

Jacco commented:
@Keraito thx for the review

Changed a few of the suggestions slightly.

Only three thing left to discuss:
[https://github.com/reactjs/nl.reactjs.org/pull/44/files](https://github.com/reactjs/nl.reactjs.org/pull/44/files)

Keraito commented:
Looks good, need to trigger a rebuild of some kind and then merge

Jacco commented:
> Looks good, need to trigger a rebuild of some kind and then merge

You mean the ci/circle build?

Keraito commented:
Yes

Jacco commented:
For the sync PRs it runs automayically...
